module/attribute-date/src/Domain/Entity/AbstractDateAttribute.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\AttributeDate\Domain\Entity;
+
+use Ergonode\Attribute\Domain\Entity\AbstractAttribute;
+use Ergonode\Attribute\Domain\Entity\AttributeId;
+use Ergonode\Attribute\Domain\Event\Attribute\AttributeParameterChangeEvent;
+use Ergonode\Attribute\Domain\ValueObject\AttributeCode;
+use Ergonode\AttributeDate\Domain\ValueObject\DateFormat;
+use Ergonode\Core\Domain\ValueObject\TranslatableString;
+use JMS\Serializer\Annotation as JMS;
+
+/**
+ */
+abstract class AbstractDateAttribute extends AbstractAttribute
```

Comment:
Abstract of abstract and next abstract will be OOP killer :) Maybe better solution exists?

Reply:
Application need to be open for new types of attributes and and their management.  In many services we have to recognize an attribute class and deal with it properly (grid presentation, data validation, etc). 
Creating new attributes on the specific classes causes that we have to handle it in the same way, it is enough in mouse cases, but unfortunately not always. 
Based on Abstract classes gives more flexibility in this matter.

Bye if you have other/better solution we are open for suggestions.